### PR TITLE
Fix a crash which happens when user shakes the phone #ANDROID-15102

### PR DESCRIPTION
**Pull request by juangardi21**
### :goal_net: What's the goal?
Fix a crash which happens when user shakes the phone, tweaks activity is shown, goes back and then shakes again.

### :construction: How do we do it?
* _Ensure navigation happens when the state is in the proper lifecycle: Instead of calling navigate directly from the shake detector, check the NavController state to make sure it's ready to navigate._
* _Use a lifecycle-aware state: Try handling navigation in response to a state within the Composable flow, rather than inside LaunchedEffect_
* _Create new function to not depends on NavController_

### :blue_book: Documentation changes?
- [x] README updated

### :test_tube: How can I test this?
- [x] Before:

https://github.com/user-attachments/assets/fd94b047-52d6-4073-a168-7b223317102d

- [x] After:

https://github.com/user-attachments/assets/f4bcc859-1385-4ad1-b3da-a51501d2971b

### CODE REVIEW

File: library/src/enabled/java/com/telefonica/tweaks/Tweaks.kt [hunk 1]
```diff
@@ -94,14 +96,18 @@ open class Tweaks : TweaksContract {
 @Composable
 fun NavController.navigateToTweaksOnShake() {
```
Comment (jdelga) [inline library/src/enabled/java/com/telefonica/tweaks/Tweaks.kt]: This feature depends on a NavController, although in some cases we might not have a NavController.

In B2P we have this launched effect that does not depend on it, and also allows a custom action to launch the tweaks:

https://github.com/Telefonica/b2p-germany-android/blob/4a9f90436777833358e891f4864d75ad9308309e/app/src/debug/java/com/telefonica/germany/b2p/core/tweaks/ShakeDetectorWrapper.kt#L18

I think this approach would be very helpful to open tweaks from the command line into a different activity (same as SmartWifi does)

Comment (juangardi21) [reply library/src/enabled/java/com/telefonica/tweaks/Tweaks.kt]: I've created another function which does not depends on navcontroller

File: library/src/enabled/java/com/telefonica/tweaks/Tweaks.kt [hunk 2]
```diff
@@ -90,12 +90,24 @@ open class Tweaks : TweaksContract {
             component.inject(reference)
         }
     }
+}
 
+@Composable
+fun NavController.navigateToTweaksOnShake() {
+    DetectShakeAndNavigate {
+        navigate(TWEAKS_NAVIGATION_ENTRYPOINT)
```
Comment (jdelga) [inline library/src/enabled/java/com/telefonica/tweaks/Tweaks.kt:98]: You can set this as default value for `onOpenTweaks` and leave only one method

File: library/src/enabled/java/com/telefonica/tweaks/Tweaks.kt (hunk 1 above)
Comment (jdelga) [reply library/src/enabled/java/com/telefonica/tweaks/Tweaks.kt]: Cool! Thanks!

File: library/src/enabled/java/com/telefonica/tweaks/Tweaks.kt (hunk 2 above)
Comment (juangardi21) [reply library/src/enabled/java/com/telefonica/tweaks/Tweaks.kt:98]: I don't think I can, because navigate is callable only within the NavController scope

Comment (jdelga) [reply library/src/enabled/java/com/telefonica/tweaks/Tweaks.kt:98]: Ok I see, one method depends on the NavController and the other one doesn't

**Review by jdelga — APPROVED** [review]

**Review by DevPabloGarcia — APPROVED** [review]
Nice catch!

**Review by dpastor — APPROVED** [review]

**Pull request merged**